woocommerce/woocommerce
# Code Freeze CLI: Add branch and milestone commands to Code Freeze action

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Integrate the recently added CLI commands [milestone](https://github.com/woocommerce/woocommerce/pull/37723) and [branch](https://github.com/woocommerce/woocommerce/pull/37914) to the code freeze action.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Prerequisites
1. You must have a `GITHUB_TOKEN` environment variable available in your PATH. This needs to be set to a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) with repo permissions enabled.
2. Fork woocommerce onto your own namespace. Push this branch to your fork

#### Test the action
1. Get `trunk` up to date on fork as well be working with the code freeze for 7.8
3. Make a 7.7.0 tagged release on fork
4. Delete `release/7.8` branch on fork if it already exists
5. Delete `prep/trunk-for-next-dev-cycle-7.9` on your fork if it already exists
6. Delete `update/7.8-changelog` on your fork if it already exists
7. Delete `delete/7.8-changelog` on your fork if it already exists
8. Delete 7.9.0 milestone if it already exists
9. Run the `Release: Code freeze` action from branch `update/code-freeze-milestone-branch`. You'll want to add a time override so the action runs because its not code freeze day. `2023-04-17T15:00:00.000` works well.
<img width="383" alt="image" src="https://user-images.githubusercontent.com/1922453/234181260-3bdcd95e-37e4-4aae-a7ae-f036bb881da9.png">
10. Make sure the action passes along with `Release: Generate changelog` which is kicked off by the process.
11. You should have three resulting PRs
<img width="443" alt="image" src="https://user-images.githubusercontent.com/1922453/234457086-a7a30e4e-97f7-4dee-ad88-11d9884ebb9c.png">
